graph/graph.go

```diff
@@ -27,11 +28,64 @@ import (
 	"github.com/docker/docker/runconfig"
 )
 
+type retainedLayers struct {
```

Comment:
Can we have a comment on this type explaining what the semantic meaning is of the keys and value slices in this map.